node.go

```diff
@@ -63,11 +69,11 @@ func (n *Node) _bucketSize() int {
 	// gets coalesced to -1.
 	// We also prevent overflows and underflows, as we convert int64 to int.
 	const maxBucketSize = 1 << 20 // 1MiB
```

Comment:
oh, here's a max, but oh boy it's big! we might have to revisit that later, and I don't see an equivalent here for bitWidth which could have worse effects.

Reply:
Captured https://github.com/ipld/go-ipld-adl-hamt/issues/40